src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/CompoundTargetFrameworkValueProvider.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties
+{
+    /// <summary>
+    /// The complex Target Framework property consists of three UI-exposed properties: target framework, target OS and target OS version.
+    /// Those combine and produce a single value like this: [target framework]-[target OS][target OS version].
+    /// This class intercepts those values from the UI to saved them into the TargetFramework property, as well as retrive those values from that same property.
+    /// </summary>
+    [ExportInterceptingPropertyValueProvider(
+        new[]
+        {
+            TargetFrameworkProperty,
+            TargetOSProperty,
```

Comment:
I think calling these different from the underlying values they represent; TargetPlatformIdentifier and TargetPlatformVersion is just resulting in confusion. While they get persisted in TargetFramework user property; they get translated to TargetPlatformXXX properties.

Reply:
I agree that naming consistency helps bring clarity into the code. At the end, I went ahead and changed every reference to those properties to a platform-related name (and follow the naming of existing associated properties), and kept the OS references in the actual string being shown in the UI. Hope this helps :)